### PR TITLE
add support for http digest auth between proxy

### PR DESCRIPTION
**Pull request by luozhaoyu**
https://github.com/sigmavirus24/requests-toolbelt/issues/53

### CODE REVIEW

File: requests_toolbelt/auth.py [hunk 1]
```diff
@@ -64,3 +66,69 @@ def __call__(self, request):
...
         request.register_hook('response', self.handle_401)
         return request
+
+
+class HTTPProxyDigestAuth(HTTPDigestAuth):
+    """HTTP digest authentication between proxy"""
+    def __init__(self, *args, **kwargs):
+        """
+        Attributes:
+            stale_rejects: The number of rejects indicate that:
```
Comment (sigmavirus24) [inline requests_toolbelt/auth.py]: First bit of feedback: Please make the doc-strings conform to the style (sphinx) of the rest of the library. We don't document `__init__` but this documentation about initialization parameters that are different from `requests.auth.HTTPDigestAuth` is appropriate for the class level docstring.

Comment (luozhaoyu) [inline requests_toolbelt/auth.py]: Thank for pointing it out. Already correct it.

**Comment by sigmavirus24** [comment]
@luozhaoyu sorry I never noticed you updated this. I'll have to look soon and rebase it myself for you.

**Comment by sigmavirus24** [comment]
@luozhaoyu I've rebased this in #63. It'll be released in 0.4.0. Thanks! :sparkles: :cake: :sparkles:

**Comment by sigmavirus24** [comment]
@luozhaoyu how can I give you credit in the release notes?

**Comment by luozhaoyu** [comment]
@sigmavirus24 Well, sorry I am not sure what you are meaning... Maybe you write some "add support for proxy digest authentication" in the release notes?

**Comment by sigmavirus24** [comment]
@luozhaoyu I meant, what name should I use to credit your contribution to?

**Comment by luozhaoyu** [comment]
@sigmavirus24 haha~ I see, please use the name "Zhaoyu Luo" and the email address is luozhaoyu90@gmail.com~

**Comment by sigmavirus24** [comment]
@luozhaoyu done in 2e0883736bd864b9e80ea7fd0923c6513d6bf7a6

**Pull request closed without merging**